onmt/Models.py

```diff
@@ -676,10 +482,7 @@ def init_decoder_state(self, src, context, enc_hidden):
         if self.decoder.decoder_type == "transformer":
             return TransformerDecoderState(src)
         elif self.decoder.decoder_type == "cnn":
-            init_state = CNNDecoderState()
-            scale_weight = 0.5 ** 0.5
-            init_state.init_src = (context + enc_hidden) * scale_weight
-            return init_state
+            return CNNDecoderState(context, enc_hidden)
```

Comment:
Not sure where this method belongs, but probably should be associated with each decoder not NMT model. Can you use this PR to move these functions to each Decoder?

Reply:
Make sense. Will do.